src/lib/openapi/spec/context-fields-schema.ts

```diff
@@ -0,0 +1,19 @@
+import { FromSchema } from 'json-schema-to-ts';
+import { contextFieldSchema } from './context-field-schema';
+import { legalValueSchema } from './legal-value-schema';
+
+export const contextFieldsSchema = {
+    $id: '#/components/schemas/contextFieldsSchema',
+    type: 'array',
+    items: {
+        $ref: '#/components/schemas/contextFieldSchema',
+    },
+    components: {
+        schemas: {
+            contextFieldSchema,
+            legalValueSchema,
```

Comment:
It's interesting that I needed to add this here to get the correct typing in the end, even though it's already being referenced inside `contextFieldSchema`. Something to take note of.

Reply:
Yeah, needs all nested schemas. Would be nice to avoid that requirement.